x-pack/legacy/plugins/lens/public/indexpattern_plugin/operations/definitions/column_types.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Operation, DimensionPriority } from '../../../types';
+
+/**
+ * This is the root type of a column. If you are implementing a new
+ * operation, extend your column type on `BaseIndexPatternColumn` to make
+ * sure it's matching all the basic requirements.
+ */
+export interface BaseIndexPatternColumn extends Operation {
+  // Private
+  operationType: string;
+  suggestedPriority?: DimensionPriority;
+}
+
+type Omit<T, K> = Pick<T, Exclude<keyof T, K>>;
```

Comment:
Good that you mention it, it's not necessary anymore because the `operationType` of `BaseIndexPatternColumn` is only a `string` and in that case the intersection type works correctly. I removed the omit and added a comment.